galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/OidcProvider.java

```diff
@@ -203,25 +207,32 @@ public JsonArray getJsonWebKeysFromIssuer(String issuer) throws IOException, Int
         HttpResponse<String> response = httpClient.send(getRequest, BodyHandlers.ofString());
 
         JsonObject responseBodyJson = gson.fromJson(response.body(), JsonObject.class);
-        if (!responseBodyJson.has("keys")) {
-            logger.error("Error: No JSON Web Keys were found at the '" + issuer + "/keys' endpoint.");
-            return null;
+        JsonArray updatedJwks = null;
+        if (responseBodyJson.has("keys")) {
+            logger.info("Successfully retrieved JSON Web Keys from issuer");
+            updatedJwks = responseBodyJson.get("keys").getAsJsonArray();
+        } else {
+            logger.error("Error: No JSON Web Keys were found at the '" + issuerUrl + "/keys' endpoint.");
         }
-        return responseBodyJson.get("keys").getAsJsonArray();
+        return updatedJwks;
     }
 
     /**
      * Gets a JSON Web Key with a given key ID ('kid') from an OpenID connect issuer's /keys endpoint, returned as a JSON object
      */
-    public JsonObject getJsonWebKeyFromIssuerByKeyId(String keyId, String issuer) throws IOException, InterruptedException {
-        JsonArray jsonWebKeys = getJsonWebKeysFromIssuer(issuer);
+    public JsonWebKey getJsonWebKeyByKeyId(String keyId) throws IOException, InterruptedException {
+        // Check if it is time to refresh the cached JSON web keys
+        if (jsonWebKeys == null || nextJwkRefresh.isBefore(Instant.now())) {
```

Comment:
I see you are using Mockito. I guess there's nothing wrong with this approach... Personally I prefer to avoid that, as it's yet another dependency we could avoid, amongst other things.